partiql-parser/src/scanner.rs

```diff
@@ -144,10 +170,20 @@ impl<'val> PartiQLScanner<'val> {
                     Rule::QuotedIdentifier => {
                         Content::Identifier(normalize_quoted_ident(ident_pair.as_str()))
                     }
-                    _ => return ident_pair.syntax_error(),
+                    _ => return ident_pair.unexpected(),
+                }
+            }
+            Rule::Number => {
+                let number_pair = pair.into_inner().exactly_one()?;
+                match number_pair.as_rule() {
+                    Rule::Integer => Content::IntegerLiteral(parse_num(number_pair)?),
+                    Rule::Decimal | Rule::DecimalExp => {
+                        Content::DecimalLiteral(parse_num(number_pair)?)
+                    }
```

Comment:
This is where we use that generic `parse_num` to fill in the literal, we leverage type inference to select `BigInt` and `BigDecimal` as appropriate.